app/services/concerns/xccdf/rule_references.rb

```diff
@@ -7,30 +7,15 @@ module RuleReferences
 
     included do
       def save_rule_references
-        @rule_references = new_rule_references + existing_rule_references
-
-        ::RuleReference.import!(new_rule_references)
-      end
-
-      private
-
-      def new_rule_references
-        @new_rule_references ||= new_op_rule_references.map do |op_reference|
-          RuleReference.new(href: op_reference.href, label: op_reference.label)
+        @rule_references = @op_rule_references.map do |op_rr|
+          ::RuleReference.new(href: op_rr.href, label: op_rr.label)
         end
-      end
 
-      def existing_rule_references
-        @existing_rule_references ||= ::RuleReference
-                                      .find_unique(@op_rule_references)
-      end
-
-      def new_op_rule_references
-        @existing_href_labels = existing_rule_references.pluck(:href, :label)
-
-        @op_rule_references.reject do |op_reference|
-          @existing_href_labels.include? [op_reference.href, op_reference.label]
-        end
+        ::RuleReference.import!(@rule_references,
+                                on_duplicate_key_update: {
+                                  conflict_target: %i[label href],
+                                  columns: %i[label href]
```

Comment:
I'm not sure if `on_duplicate_key_update` is necessary here because if `label` and `href` are the target columns, that would mean that if there's another record with the same values for those 2 columns, it would update the `label` and `href` which would just make them the same thing, I think. So I think this is the same thing as just doing `ignore: true` ?

Reply:
It is, even if it "doesn't do anything" :see_no_evil: I know it sounds weird, but the point here is to update the IDs of all objects in the `@rule_references` array. Notice that I removed the `find_or_initialize_by` to avoid a bunch of `find_by` calls to our database. If I'd do an import with `ignore: true`, the `@rule_references` array would only have updated IDs for the newly created entities.